image2display: init at 1.1.6.0

bubblepipe opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Tert0 commented:
Some suggestions:
```diff
diff --git a/pkgs/by-name/im/image2display/package.nix b/pkgs/by-name/im/image2display/package.nix
index b3730f2f49cc..31bbe1a5f338 100644
--- a/pkgs/by-name/im/image2display/package.nix
+++ b/pkgs/by-name/im/image2display/package.nix
@@ -1,8 +1,9 @@
 {
-  pkgs,
   lib,
   buildDotnetModule,
   fetchFromGitHub,
+  dotnet-sdk_9,
+  nix-update-script,
 }:
 
 let
@@ -21,16 +22,17 @@ buildDotnetModule {
 
   nugetDeps = ./deps.json;
 
-  dotnet-sdk = pkgs.dotnet-sdk_9;
+  dotnet-sdk = dotnet-sdk_9;
 
   projectFile = "Image2Display/Image2Display/Image2Display.csproj";
 
-  meta = with lib; {
+  passthru.updateScript = nix-update-script {};
+
+  meta = {
+    description = "Cross-platform image and font data processing tool for generating data usable by microcontrollers";
     homepage = "https://github.com/chenxuuu/image2display/blob/master/README_EN.md";
-    description = "A cross-platform image and font data processing tool for generating data usable by microcontrollers.";
-    license = licenses.mit;
-    platforms = platforms.all;
-    mainProgram = "Image2Display";
+    license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ bubblepipe ];
+    mainProgram = "Image2Display";
   };
 }
```

Please make a separate commit for adding yourself to the mainainers list.

Also I think `libHarfBuzzSharp.so` and `libSkiaSharp.so` should not be in `bin/`.

bubblepipe committed:
add maintainer

bubblepipe commented:
> ```diff
> index b3730f2f49cc..31bbe1a5f338 100644
> ```

Thank you very much for the suggestions! I have attampted to delete the `lib` files but I am not sure if I have done it properly?

SuperSandro2000 commented on pkgs/by-name/im/image2display/package.nix:
```diff
@@ -0,0 +1,45 @@
+
+  meta = {
+    homepage = "https://github.com/chenxuuu/image2display/blob/master/README_EN.md";
+    description = "A cross-platform image and font data processing tool for generating data usable by microcontrollers.";
```
```suggestion
    description = "Cross-platform image and font data processing tool for generating data usable by microcontrollers";
```

SuperSandro2000 commented on pkgs/by-name/im/image2display/package.nix:
```diff
@@ -0,0 +1,45 @@
+  projectFile = "Image2Display/Image2Display/Image2Display.csproj";
+
+  postFixup = ''
+    rm $out/bin/libAvaloniaNative.*
+    rm $out/bin/libHarfBuzzSharp.*
+    rm $out/bin/libSkiaSharp.*
```
One of those rm is not matching any files

bubblepipe42 replied:
On nix darwin there is an additional `libAvaloniaNative.dylib`, compared with linux systems. 🤔
Should be fixed by using `rm -f`

SuperSandro2000 replied:
Not a fan of this as we could accumulate dead code through this. How about we make it conditional based on darwin?

bubblepipe42 replied:
Sure! 
This should be better: 
```
  postFixup =
    lib.optionalString stdenv.isLinux ''
      rm $out/bin/libHarfBuzzSharp.so
      rm $out/bin/libSkiaSharp.so
    ''
    + lib.optionalString stdenv.isDarwin ''
      rm $out/bin/libAvaloniaNative.dylib
      rm $out/bin/libHarfBuzzSharp.dylib
      rm $out/bin/libSkiaSharp.dylib
    '';
```

bubblepipe replied:
Sure! 
```
  postFixup =
    lib.optionalString stdenv.isLinux ''
      rm $out/bin/libHarfBuzzSharp.so
      rm $out/bin/libSkiaSharp.so
    ''
    + lib.optionalString stdenv.isDarwin ''
      rm $out/bin/libAvaloniaNative.dylib
      rm $out/bin/libHarfBuzzSharp.dylib
      rm $out/bin/libSkiaSharp.dylib
    '';
```
This should be better

bubblepipe committed:
image2display: init at 1.1.6.0

Cross-platform image and font data processing tool for generating data usable by microcontrollers
